flash/vision/backbones.py

```diff
@@ -51,6 +52,24 @@
 OBJ_DETECTION_BACKBONES = FlashRegistry("backbones")
 
 
+def catch_url_error(fn):
+
+    @functools.wraps(fn)
+    def wrapper(*args, pretrained=False, **kwargs):
+        try:
+            return fn(*args, pretrained=pretrained, **kwargs)
+        except urllib.error.URLError:
+            result = fn(*args, pretrained=False, **kwargs)
+            rank_zero_warn(
+                "Failed to download pretrained weights for the selected backbone. The backbone has been created with"
+                " `pretrained=False` instead. If you are loading from a local checkpoint, this warning can be safely"
+                " ignored.", UserWarning
+            )
```

Comment:
Changed the order so if the fn call fails too, the warning is still shown

```suggestion
            rank_zero_warn(
                "Failed to download pretrained weights for the selected backbone. The backbone has been created with"
                " `pretrained=False` instead. If you are loading from a local checkpoint, this warning can be safely"
                " ignored.", UserWarning
            )
            result = fn(*args, pretrained=False, **kwargs)
```

Reply:
Not sure we want this. If it fails woth `pretrained=False` then the warning is no longer true when it says "the backbone has been created"?